cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+    private static String destTableDDL=new String();           //contains destination table ddl
+    private static String filterCondition=new String();
+    private static String srcStgTableLocation=new String();
+    private static String destTableLocation=new String();
+    private static final String[][] PARAMS_STRUCTURE = {
+            {"pp", "parent-process-id", "Parent Process id of migration preprocessor"},
+            {"p", "process-id", " Process id of migration preprocessor"},
+            {"ied", "instance-exec-id", " instance exec id of preprocessor"}
+    };
+
+
+    protected static Connection getHiveJDBCConnection(String dbName, String hiveConnection) throws Exception {
+        try {
+            Class.forName(IMConstant.HIVE_DRIVER_NAME);
+            String hiveUser = IMConfig.getProperty("etl.hive-jdbcuser");
+            String hivePassword = IMConfig.getProperty("etl.hive-jdbcpassword");
+            Connection con = DriverManager.getConnection(hiveConnection + "/" + dbName, hiveUser, hivePassword);
+            con.createStatement().execute("set hive.exec.dynamic.partition.mode=nonstrict");
+            con.createStatement().execute("set hive.exec.dynamic.partition=true");
+            con.createStatement().execute("set hive.exec.max.dynamic.partitions.pernode=1000");
+            return con;
+        } catch (ClassNotFoundException e) {
+            throw new Exception(e);
+        } catch (SQLException e) {
+            throw new Exception(e);
+        }
+    }
+
+    public MigrationPreprocessorInfo execute(String[] params) throws Exception{
+
+        CommandLine commandLine = getCommandLine(params, PARAMS_STRUCTURE);
+        String parentProcessId=commandLine.getOptionValue("parent-process-id");
+        String processId = commandLine.getOptionValue("process-id");
+        String instanceExecId = commandLine.getOptionValue("instance-exec-id");
+        MigrationPreprocessorInfo mpInfo = prepareMigrate(parentProcessId,processId,instanceExecId);
+        return mpInfo;
+    }
+    private MigrationPreprocessorInfo prepareMigrate(String parentProcessId, String processId, String instanceExecId) throws Exception {
+        MigrationPreprocessorInfo migrationPreprocessorInfo = new MigrationPreprocessorInfo();
+        Properties params=getParams(parentProcessId,"hive-migration");
+        String table = params.get("src-table").toString();
+        String sourceDb = params.get("src-db").toString();
+        String destDb= params.get("dest-db").toString();
+        String sourceStgtable=table + "_stg";
+        String bdreTechPartition=params.get("bdre-tech-pt").toString();
+        String sourceNameNodeAddress=params.get("src-nn").toString();
+        String destNameNodeAddress=params.get("dest-nn").toString();
+        String sourceJobTrackerAddress=params.get("src-jt").toString();
+        String destJobTrackerAddress=params.get("dest-jt").toString();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "destJobTrackerAddress" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1481)